### PR TITLE
Updated the documentation for the terrain and furniture entries.

### PR DESCRIPTION
**Pull request by BevapDin**
Someone should probably look at the text for grammar and stuff. And of course for actual correctness. And maybe someone wants to fill in the TODOs, or maybe later or so.

### CODE REVIEW

File: doc/JSON_INFO.md [hunk 1]
```diff
@@ -831,35 +831,198 @@ The format also support snippet ids like above.
...
+    "id": "f_toilet",
+    "name": "toilet",
+    "symbol": "&",
+    "color": "white",
+    "move_cost_mod": 2,
+    "required_str": 18,
+    "flags": ["TRANSPARENT", "BASHABLE", "FLAMMABLE_HARD"],
+    "crafting_pseudo_item": "anvil",
+    "examine_action": "toilet",
+    "close": "f_foo_closed",
```
Comment (Coolthulhu) [inline doc/JSON_INFO.md]: `foo`? More like `loo`!

File: doc/JSON_INFO.md [hunk 2]
```diff
@@ -831,35 +831,198 @@ The format also support snippet ids like above.
...
+Fixed string, must be "terrain" to identify the JSON object as such.
+
+#### "id", "name", "symbol", "color", "bgcolor", "max_volume", "open", "close", "bash", "deconstruct", "examine_action", "flags"
+Same as for furniture, see below in the chapter "Common to furniture and terrain".
+
+#### "move_cost"
+Move cost to move through. A value of 0 means it's impassable (e.g. wall). You should not use negative values. The positive value is multiple of 50 move points, e.g. value 2 means the player uses 2*50 = 100 move points when moving across the terrain.
+
+#### "trap"
+(Optional) Id of the build-in trap of that terrain. For example the terrain "t_pit" has the build in trap "tr_pit". Every tile in the game that has the terrain "t_pit" has therefor also an implicit trap "tr_pit" on it. Both are inseparable (the player can not deactivate the build-in trap, but changing the terrain will also deactivate the built-in trap).
```
Comment (Coolthulhu) [inline doc/JSON_INFO.md]: Could use an extra note about non-terrain and terrain traps being exclusive

File: doc/JSON_INFO.md [hunk 3]
```diff
@@ -831,35 +831,198 @@ The format also support snippet ids like above.
...
+The terrain / furniture that will be set when the original is destroyed. This is mandatory for bash entries in terrain, but optional for entries in furniture (it defaults to no furniture).
+
+#### "explosive"
+TODO
+
+#### "collapse_radius"
+TODO
+
+#### "destroy_only"
+TODO
```
Comment (Coolthulhu) [inline doc/JSON_INFO.md]: This one might be obsolete. I can't find it anywhere in the jsons.

Comment (BevapDin) [inline doc/JSON_INFO.md]: It's still used in map.cpp:3315

File: doc/JSON_INFO.md [hunk 4]
```diff
@@ -831,35 +831,198 @@ The format also support snippet ids like above.
...
+    "explosive": 1,
+    "collapse_radius": 2,
+    "destroy_only": true,
+    "bash_below": true,
+    "tent_centers": ["f_groundsheet", "f_fema_groundsheet", "f_skin_groundsheet"],
+    "items": "bashed_item_result_group"
+}
+```
+
+#### "str_min", "str_max", "str_min_blocked", "str_max_blocked", "str_min_supported", "str_max_supported", "str_min_roll", "str_max_roll"
```
Comment (BevapDin) [inline doc/JSON_INFO.md]: str_max_roll and str_min_roll are not used anywhere in the code. Shall I remove them from the `map_bash_info`? Or are the supposed to come back?

Comment (Coolthulhu) [inline doc/JSON_INFO.md]: I don't think anyone will miss them. I don't even understand what were they supposed to do.

Comment (kevingranade) [inline doc/JSON_INFO.md]: They were parameters for bash resistance that were either refactored away
or never implemented.

**Pull request merged**